processing/src/main/java/org/apache/druid/segment/vector/VectorColumnSelectorFactory.java

```diff
@@ -62,16 +66,29 @@ default int getMaxVectorSize()
   SingleValueDimensionVectorSelector makeSingleValueDimensionSelector(DimensionSpec dimensionSpec);
 
   /**
-   * Returns a string-typed, multi-value-per-row column selector. Should only be called on columns where
-   * {@link #getColumnCapabilities} indicates they return STRING. Unlike {@link #makeSingleValueDimensionSelector},
-   * this should not be called on nonexistent columns.
+   * Returns a dictionary encoded, string-typed, multi-value-per-row column selector. Should only be called on columns
+   * where {@link #getColumnCapabilities} indicates they return STRING. Unlike
+   * {@link #makeSingleValueDimensionSelector}, this should not be called on nonexistent columns.
    *
    * If you need to write code that adapts to different input types, you should write a
    * {@link org.apache.druid.segment.VectorColumnProcessorFactory} and use one of the
    * {@link org.apache.druid.segment.ColumnProcessors#makeVectorProcessor} functions instead of using this method.
    */
   MultiValueDimensionVectorSelector makeMultiValueDimensionSelector(DimensionSpec dimensionSpec);
 
+  /**
+   * Returns a dimension object selector. Useful for non-dictionary encoded strings, or when using the dictionary
+   * isn't useful. Right now, this should probably only be called on single valued STRING inputs (multi-value STRING
```

Comment:
Get rid of the "probably" 🙂

Either it will work or it won't. If it doesn't work, say firmly not to do it, and ideally describe what will happen if you do it accidentally (the method will throw an error? the method will return a selector with undefined behavior? etc). If it does work, but might be a bad idea for some reason, spell that out.

Reply:
i got rid of this method, and merged these javadocs into the `makeObjectSelector` javadocs